pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -129,21 +129,57 @@ func (i *KumaInjector) needInject(pod *kube_core.Pod, ns *kube_core.Namespace) (
 		log.V(1).Info("pod fulfills exception requirements")
 		return false, nil
 	}
-	enabled, exist, err := metadata.Annotations(pod.Annotations).GetEnabled(metadata.KumaSidecarInjectionAnnotation)
+
+	for _, container := range pod.Spec.Containers {
+		if container.Name == util.KumaSidecarContainerName {
+			log.V(1).Info("pod already has Kuma sidecar")
+			return false, nil
+		}
+	}
+
+	enabled, exist, err := metadata.Annotations(pod.Labels).GetEnabled(metadata.KumaSidecarInjectionAnnotation)
 	if err != nil {
 		return false, err
 	}
 	if exist {
+		if !enabled {
+			log.V(1).Info("pod has kuma.io/sidecar-injection: disabled label")
```

Comment:
```suggestion
			log.V(1).Info("pod has %q label", "kuma.io/sidecar-injection: disabled")
```

Reply:
%q is not valid in log, but added quotes manually ;)